src/promptix/core/components/prompt_loader.py

```
@@ -0,0 +1,129 @@
+            return self._prompts
+            
+        try:
+            # Check for unsupported JSON files first
+            unsupported_files = config.check_for_unsupported_files()
+            if unsupported_files:
+                json_file = unsupported_files[0]  # Get the first JSON file found
+                raise UnsupportedFormatError(
+                    file_path=str(json_file),
+                    unsupported_format="json",
+                    supported_formats=["yaml"]
+                )
```

Ruff (0.12.2)

Abstract `raise` to an inner function

(TRY301)